frontend/src/models/dashboardsModel.tsx

```diff
@@ -234,22 +235,40 @@ export const dashboardsModel = kea<dashboardsModelType>({
     },
 
     selectors: ({ selectors }) => ({
+        nameCompareFunction: [
+            () => [() => 1],
+            () => (a: DashboardBasicType, b: DashboardBasicType) => {
+                const firstName = a.name ?? 'Untitled'
+                const secondName = b.name ?? 'Untitled'
+
+                if (
+                    firstName.startsWith(GENERATED_DASHBOARD_PREFIX) &&
+                    !secondName.startsWith(GENERATED_DASHBOARD_PREFIX)
+                ) {
+                    return 1 // a should come after b
+                }
+                if (
+                    !firstName.startsWith(GENERATED_DASHBOARD_PREFIX) &&
+                    secondName.startsWith(GENERATED_DASHBOARD_PREFIX)
+                ) {
+                    return -1 // a should come before b
+                }
+
+                return firstName.localeCompare(secondName)
+            },
```

Comment:
We're gonna do a lot of these `startsWith` twice, which also makes this harder to read. Let's do some `firstNameIsGenerated` consts instead